### ISSUE TITLE
PC-98 FM: Audible differences between Linux and Windows builds

### ISSUE THREAD

**Issue by joncampbell123** [joncampbell123/dosbox-x#563]
The percussion sounds different between Linux and Windows, for some reason.

I'm not sure if this is related to the order of FM initialization, or uninitialized variables.

Notice the Linux version makes the percussion more audible, but panned off to the left.

[th05_music_linux_windows_audible_differences.zip](https://github.com/joncampbell123/dosbox-x/files/1764407/th05_music_linux_windows_audible_differences.zip)

**Comment by leonier** [comment]
According to this https://www.bilibili.com/video/av11486485/ captured from a real PC-9801US, the percussion ("rhythm") part of TH05's music is indeed used OPNA stereo and panned off to the left if  PMDB2.COM driver is used.

**Comment by joncampbell123** [comment]
Is it only the percussion? The FM music, besides the percussion, sounds like it's strictly monaural.

**Comment by leonier** [comment]
Yes the FM and PSG channels are strictly in the middle. Without Rhythm channels TH2-TH5 is strictly monaural.

**Comment by joncampbell123** [comment]
Sounds great in Linux now, but for some odd reason, Windows builds do not render the rhythm section in stereo.

**Comment by leonier** [comment]
Why don't you use macros (_MSC_VER, __MINGW32__) for that code in pc98_fm.cpp? The problem might lie in compiler optimization of Microsoft compilers.
 Do rhythm channels sound different when you use -O1, -O0 and disable link-time code generation?

**Comment by joncampbell123** [comment]
I'm sorry I wasn't clear in the post.

The cause for the different sound is entirely my fault because I was testing on a Windows setup where I had forgotten to copy the rhythm WAV files into the same directory, instead of the other Windows machine where I had already installed them.

Having installed those WAV files, both Linux and Windows produce the same rendition of the rhythm section now.

**Comment by leonier** [comment]
I could post a recording from my MinGW Windows build for you to compare.
[op_000.zip](https://github.com/joncampbell123/dosbox-x/files/1770345/op_000.zip)

**Comment by joncampbell123** [comment]
That's what it's supposed to sound like. The percussion instruments that are panned to the left are rendered the way they're supposed to. TH05 likes to pan rhythm instruments to the left or right for some reason.

I initially filed this issue because Windows builds (due to the mistake I just mentioned) sounded different due to missing rhythm instruments. I fixed that by installing the files.
